app/services/data_api/tad_application_export.rb

```diff
@@ -89,5 +92,35 @@ def nationalities
     def concatenate(array)
       array.to_a.join('|')
     end
+
+    def offer_originally_deferred_at
+      return if application_choice.offer_deferred_at.nil?
+
+      # We get the audits in the order they have been created, first record is the oldest
+      old_attributes = audits.map(&:old_attributes)
+      # [nil, nil, timestamp]
+      first_deferred_at = old_attributes.map do |attributes|
+        attributes[:offer_deferred_at]
+      end.compact.first
+
+      original_deferred_at = first_deferred_at || application_choice.offer_deferred_at
+      original_deferred_at.to_time.iso8601
+    end
+
+    def application_choice_deferred_confirmed_at
+      return if application_choice.offer_deferred_at.nil?
+
+      audit_when_deferred_offer_has_been_confirmed = audits.to_a.select do |audit|
+        audit.audited_changes['status'] == %w[offer_deferred recruited]
```

Comment:
No, if it's an update. The array will always have 2 elements. This is because audited each change to a column as an event, `before and after`.

This logic won't cover the scenario when an application choice changes from `offer_deferred` to `conditions not met` to `recruited` for example. If we mess about in the console and the change event is not from `offer_deferred` to `recruited` we won't show any timestamp for `application_choice_deferred_confirmed_at`

Reply:
Thanks for clarifying!